src/Components/api.js

```diff
@@ -1,5 +1,7 @@
 const rootUrl = 'https://node-course-e-commerce-8r2s.onrender.com'
 
+// AKOS REVIEW NOTE: why duplicate the logic within each controller, if we have them here in the api module?
```

Comment:
For example in `src/pages/Login.js` we could do the following:

```
import { handleLogin } from "../../Components/api"

// delete 11 to 34 in Login.js


// but keep everything else:

  const handleSubmit = async (e) => {
    e.preventDefault()
    if (!email || !password) return
    const user = { email, password }
    const success = await handleLogin(user)
    if (success) {
      setEmail('')
      setPassword('')
      setRedirectToHome(true)
    }
  }
  if (redirectToHome) {
    return <Navigate to='/' />
  }
```

Reply:
You are right, it will be more structured if I make these modifications but in the component Login (line 5) I use userName,setUserName to store the name of the logged-in user (line 9)  and then it is used  in function handleLogin (line24). If I make modifications proposed , should I remove line 9 ? And  in this way,  I will use the function handleLogin from api.js , right ? I shouldn't have any conflicts as a result of this modification?

Reply:
You could move the `setUserName` call to happen inside `handleSubmit`, inside the `if (success)` conditional; since you only want to do it if the `handleLogin` worked anyway. 

So in that case, you would keep line 9, move line [24](https://github.com/akosasante/Shop-online/blob/auth/src/pages/Login.js#L24) down under line [43](https://github.com/akosasante/Shop-online/blob/auth/src/pages/Login.js#L43), and then delete lines 11-34 since you would just be importing and using the `handleLogin` from api.js.

But i have a separate comment that I'll leave below about `setUserName` so it's possible you just won't need that line at all in Login component anyway.